### ISSUE TITLE
Initialize element before running executeJs when using a renderer

### ISSUE THREAD

**Issue by tulioag** [vaadin/flow#7463]
In Combox , an attach listener is registered to initialize the connector.
```         
addAttachListener(e -> getElement().executeJs(
                "window.Vaadin.Flow.comboBoxConnector.initLazy(this)"));
```
This works in normal cases but it breaks when the combo box is used in a renderer, like in a component column of grid. It breaks in this case because the connector tries to access `this.$`, but it is undefined.

- Minimal reproducible example
```
		final TreeGrid<String> grid = new TreeGrid<>();
		grid.addComponentColumn(s -> new ComboBox());
```
 or  https://github.com/vaadin/vaadin-combo-box-flow/pull/324

- Expected behavior
`this.$` is defined when `comboBoxConnector.initLazy(this)` is called
- Actual behavior
`Cannot read property 'overlay' of undefined` in the line
` const _isItemSelected = comboBox.$.overlay._isItemSelected;`

- Versions: 
    - Vaadin / Flow version
             Tested both with flow 3.0-SNAPSHOT and platform 14.1.16.

**Comment by Legioth** [comment]
`this.$` is populated by `super.connectedCallback()`. `executeJs` is run only after any regular changes have been done to the DOM tree, but renderers break this since they are also implemented based on `executeJs`.

One idea that comes to mind would be that Flow would introduce a special version of `executeJs` that is run already earlier and then this one would be used by some specific cases such as renderers that should be effective when regular `executeJs` is run.

There's also some existing logic in the client engine for postponing `executeJs` which references elements that are not yet attached for other reasons (mainly related to deferred upgrade of custom elements in some cases). I wonder if that functionality could also be supplemented to cover renderer elements?

**Comment by centenico** [comment]
Has this been solved anyhow? Or is there a workaround this problem to avoid the errors client side?

**Comment by johannest** [comment]
This _might_ be possible to work around by postponing the execution with an extra server-browser-server roundtrip. The workaround is for TextField in Grid and for setting the aria label but it will likely work for other component as well.
```
this.getElement().executeJs("return").then(x -> {
                textField.getElement().executeJs("this.inputElement.setAttribute('aria-label', 'sometext')");
            });
```
